src/app/components/shared/date-time-filter/audio-recordings-filter.component.ts

```diff
@@ -0,0 +1,166 @@
+import { AudioRecording } from "@models/AudioRecording";
+import { Project } from "@models/Project";
+import { Region } from "@models/Region";
+import { Site } from "@models/Site";
+import { NgbDate } from "@ng-bootstrap/ng-bootstrap";
+import { fromJS } from "immutable";
+import { DateTime } from "luxon";
+import {
+  BehaviorSubject,
+  debounceTime,
+  distinctUntilChanged,
+  takeUntil,
+} from "rxjs";
+import { defaultDebounceTime } from "src/app/app.helper";
+
+export interface Model {
+  projects?: Project[];
+  regions?: Region[];
+  sites?: Site[];
+  dateFiltering?: boolean;
+  dateStartedAfter?: NgbDate;
+  dateFinishedBefore?: NgbDate;
+  timeFiltering?: boolean;
+  ignoreDaylightSavings?: boolean;
+  timeStartedAfter?: string;
+  timeFinishedBefore?: string;
+}
+
+@Component({
+  selector: "baw-audio-recordings-filter",
+  templateUrl: "audio-recordings-filter.component.html",
+  styleUrls: ["audio-recordings-filter.component.scss"],
+})
+export class AudioRecordingsFilterComponent
+  extends withUnsubscribe()
+  implements AfterViewInit, AfterContentChecked
+{
+  public constructor(private changeDetector: ChangeDetectorRef) {
+    super();
+  }
+
+  @ViewChild(NgForm) public form: NgForm;
+  @Input() public project: Project;
+  @Input() public region: Region;
+  @Input() public site: Site;
+  @Input() public constructedFilters: BehaviorSubject<Filters<AudioRecording>>;
+
+  public model: Model = { ignoreDaylightSavings: true };
+  private previousFilters = {};
+
+  public ngAfterViewInit(): void {
+    this.form.valueChanges
+      .pipe(
+        debounceTime(defaultDebounceTime),
+        distinctUntilChanged(),
+        takeUntil(this.unsubscribe)
+      )
+      .subscribe((model: Model) => this.emitFilterUpdate(model));
+  }
+
+  // TODO: Refactor the following hacky code block
+  // without this code block, the ExpressionChangedAfterItHasBeenCheckedError warning is thrown in development mode
+  public ngAfterContentChecked(): void {
+    // since we are using angular form validation & bootstrap validation, the errors attribute of the form is updated with change detection
+    // this causes the form state to update with change detection so we need to add an extra change detection cycle
+    // to ensure the updated form state (with errors) is a part of the model at end of change detection
+    this.changeDetector.detectChanges();
+  }
+
+  public emitFilterUpdate(model: Model): void {
+    // we should only send new filter requests when the user has not input any "bad" / incorrect values into the input fields
+    // e.g. 2020-31-31 is not a valid date should display an error, and not send a new filter request
+    if (!this.form.valid) {
+      return;
+    }
+
+    const [changed, newFilters] = this.generateFilters(model);
+
+    if (changed) {
+      this.constructedFilters.next(newFilters);
+    }
+  }
+
+  private generateFilters(model: Model): [boolean, Filters] {
```

Comment:
```suggestion
  private generateFilters(previous: Filters, model: Model): [boolean, Filters] {
```

embrace stateful side effects or functional methods with no side effects - don't mix the concepts